### ISSUE TITLE
Plugin crashes IntelliJ 2016 versions

### ISSUE THREAD

**Issue by voidmain** [asciidoctor/asciidoctor-intellij-plugin#135]
At first I thought this was the same issue as the JavaFX problem that has already been reported. However, I was working this morning and was using the Swing display. Plus, I had the display closed and was only editing the contents of my `adoc` file.

At this point, IntelliJ started to consume 100% of the CPU and started throwing errors continuously. It stopped responding to events and I got an error message saying that IntelliJ was unable to save settings and needed to be restarted.

I managed to capture the output for the error log though. Here's what I have:

```
Unbalanced tree. Most probably caused by unbalanced markers. Try calling setDebugMode(true) against PsiBuilder passed to identify exact location of the problem
language: Language: AsciiDoc, 
context: 'ok similar to this. If you installed somewhere other the default `PASSPORT_HOME`, your directory structure will be different, this is only for shown as an example.

[source,bash]
----
C:\inversoft\config
C:\inversoft\config\keystore
C:\inversoft\config\passport.properties
C:\inversoft\passport-backend\
----

Next, install the Windows service by changing to the directory you installed Passport (i.e. `C:\inversoft\passport-backend\apache-tomcat-8.0.12\bin`) and then running the install command. Here are those two steps:

[source,bash]
----
C:\>cd C:\inversoft\passport-backend\apache-tomcat-8.0.12\bin
C:\inversoft\passport-backend\apache-tomcat-8.0.12\bin>PassportBackend.exe /install
----

==== Service Name

As long as your installation method installed Passport as a service, the service name will be as follows:

[cols="h,m"]
|===
|Service Name
|passport-backend
|===

You can use the operating system commands to start, stop and check the status of Passport using this service name.

====

'
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:127)
	at com.intellij.lang.impl.PsiBuilderImpl.a(PsiBuilderImpl.java:1307)
	at com.intellij.lang.impl.PsiBuilderImpl.b(PsiBuilderImpl.java:1262)
	at com.intellij.lang.impl.PsiBuilderImpl.c(PsiBuilderImpl.java:1140)
	at com.intellij.lang.impl.PsiBuilderImpl.getTreeBuilt(PsiBuilderImpl.java:1124)
	at org.asciidoc.intellij.parser.AsciiDocParser.parse(AsciiDocParser.java:23)
	at com.intellij.psi.tree.ILazyParseableElementType.doParseContents(ILazyParseableElementType.java:85)
	at com.intellij.psi.tree.IFileElementType.parseContents(IFileElementType.java:43)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.b(LazyParseableElement.java:192)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:240)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:39)
	at com.intellij.psi.impl.source.text.BlockSupportImpl.a(BlockSupportImpl.java:331)
	at com.intellij.psi.impl.source.text.BlockSupportImpl.mergeTrees(BlockSupportImpl.java:277)
	at com.intellij.psi.impl.source.text.BlockSupportImpl.a(BlockSupportImpl.java:216)
	at com.intellij.psi.impl.source.text.BlockSupportImpl.reparseRange(BlockSupportImpl.java:92)
	at com.intellij.psi.impl.DocumentCommitThread.doCommit(DocumentCommitThread.java:803)
	at com.intellij.psi.impl.DocumentCommitThread$6.run(DocumentCommitThread.java:581)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1034)
	at com.intellij.psi.impl.DocumentCommitThread.a(DocumentCommitThread.java:599)
	at com.intellij.psi.impl.DocumentCommitThread.access$400(DocumentCommitThread.java:81)
	at com.intellij.psi.impl.DocumentCommitThread$2.run(DocumentCommitThread.java:415)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:494)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:443)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.psi.impl.DocumentCommitThread.a(DocumentCommitThread.java:412)
	at com.intellij.psi.impl.DocumentCommitThread.run(DocumentCommitThread.java:362)
	at com.intellij.util.concurrency.BoundedTaskExecutor.runFirstTaskThenPollAndRunRest(BoundedTaskExecutor.java:178)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$000(BoundedTaskExecutor.java:40)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:197)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)


java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:127)
	at com.intellij.lang.impl.PsiBuilderImpl.a(PsiBuilderImpl.java:1307)
	at com.intellij.lang.impl.PsiBuilderImpl.b(PsiBuilderImpl.java:1262)
	at com.intellij.lang.impl.PsiBuilderImpl.c(PsiBuilderImpl.java:1140)
	at com.intellij.lang.impl.PsiBuilderImpl.getTreeBuilt(PsiBuilderImpl.java:1124)
	at org.asciidoc.intellij.parser.AsciiDocParser.parse(AsciiDocParser.java:23)
	at com.intellij.psi.tree.ILazyParseableElementType.doParseContents(ILazyParseableElementType.java:85)
	at com.intellij.psi.tree.IFileElementType.parseContents(IFileElementType.java:43)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.b(LazyParseableElement.java:192)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:240)
	at com.intellij.psi.impl.source.text.DiffLog$ReplaceElementWithEvents.<init>(DiffLog.java:263)
	at com.intellij.psi.impl.source.text.DiffLog$ReplaceElementWithEvents.<init>(DiffLog.java:254)
	at com.intellij.psi.impl.source.text.DiffLog.appendReplaceElementWithEvents(DiffLog.java:74)
	at com.intellij.psi.impl.source.text.BlockSupportImpl.a(BlockSupportImpl.java:261)
	at com.intellij.psi.impl.source.text.BlockSupportImpl.mergeTrees(BlockSupportImpl.java:278)
	at com.intellij.psi.impl.source.text.BlockSupportImpl.a(BlockSupportImpl.java:216)
	at com.intellij.psi.impl.source.text.BlockSupportImpl.reparseRange(BlockSupportImpl.java:92)
	at com.intellij.psi.impl.DocumentCommitThread.a(DocumentCommitThread.java:953)
	at com.intellij.psi.impl.DocumentCommitThread.access$1500(DocumentCommitThread.java:81)
	at com.intellij.psi.impl.DocumentCommitThread$9.process(DocumentCommitThread.java:829)
	at com.intellij.psi.impl.DocumentCommitThread$9.process(DocumentCommitThread.java:805)
	at com.intellij.psi.impl.PsiDocumentManagerBase.a(PsiDocumentManagerBase.java:379)
	at com.intellij.psi.impl.PsiDocumentManagerBase.finishCommitInWriteAction(PsiDocumentManagerBase.java:355)
	at com.intellij.psi.impl.PsiDocumentManagerImpl.finishCommitInWriteAction(PsiDocumentManagerImpl.java:142)
	at com.intellij.psi.impl.PsiDocumentManagerBase$3.run(PsiDocumentManagerBase.java:314)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:906)
	at com.intellij.psi.impl.PsiDocumentManagerBase.finishCommit(PsiDocumentManagerBase.java:321)
	at com.intellij.psi.impl.DocumentCommitThread$7.run(DocumentCommitThread.java:636)
	at com.intellij.openapi.application.TransactionGuardImpl.a(TransactionGuardImpl.java:88)
	at com.intellij.openapi.application.TransactionGuardImpl.access$300(TransactionGuardImpl.java:40)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:113)
	at com.intellij.openapi.application.TransactionGuardImpl.submitTransaction(TransactionGuardImpl.java:123)
	at com.intellij.psi.impl.DocumentCommitThread$3.run(DocumentCommitThread.java:433)
	at com.intellij.openapi.application.TransactionGuardImpl$6.run(TransactionGuardImpl.java:291)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.a(LaterInvocator.java:326)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:310)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.a(IdeEventQueue.java:863)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:699)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:385)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

For now, I'll have to disable the plugin. Let me know if you need any other info.

**Comment by mojavelinux** [comment]
To summarize, do you think there's some sort of memory leak or infinite recursion in the AsciiDoc parser that is provided by this plugin? That seems like something that a unit test could catch.

**Comment by ahus1** [comment]
This sounds like a duplicate of #119. I suggest closing this ticket. 

I'll don't think I can fix the parser, but I can improve the error handling. I'll release a 0.14.x and 0.15.x versions shortly with at least better error handling.

**Comment by voidmain** [comment]
Yeah. I read over #119 and this does look like a duplicate. Go ahead and close this one out.

**Comment by ahus1** [comment]
@voidmain - this is fixed in 0.14.3 that has just been released. If you find more problematic Asciidoc texts please open a new ticket with a minimal example.
